Unable to cross compile ACE 7.0.2 (arm 32-bit)

### Version

The version of ACE and/or TAO you are using
ACE 7.0.2

### Host machine and operating system
Debian amd64:
uname -a:
Linux <> 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19) x86_64 GNU/Linux

### Target machine and operating system (if different from host)
Linux, gnu arm 32 bit

### Compiler name and version (including patch level)
```
root@host:~# g++ --version
g++ (Debian 8.3.0-6) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

root@host:~# arm-linux-gnueabihf-g++ --version
arm-linux-gnueabihf-g++ (Debian 8.3.0-2) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

### The $ACE_ROOT/ace/config.h file

The default implementation config_linux.h

### The $ACE_ROOT/include/makeinclude/platform_macros.GNU file

```
:~# cat ACE/include/makeinclude/platform_macros.GNU 
CROSS_COMPILE=arm-linux-gnueabihf-

ipv6=1
#xerces3=1
ssl=1

include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU
```


### Contents of $ACE_ROOT/bin/MakeProjectCreator/config/default.features

```
ipv6=1
#xerces3=1
ssl=1
versioned_namespace=1
```

### AREA/CLASS/EXAMPLE AFFECTED:

I'm unable to compile the ACE core module.

### The problem effects:

Unable to link ACE with the arm-linux-gnueabihf-g++ compiler/linker.

### Synopsis

I'm unable to cross compile ACE for 32 bit arm.

### Description

I tried the update from 7.0.1 to 7.0.2. With 7.0.2 I'm unable to compile ACE as the linker is unable to link the shared objects.
```
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/POSIX_Asynch_IO.o:(.rodata+0x2b4): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/POSIX_Asynch_IO.o:(.data.rel.ro+0x7d0): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/POSIX_CB_Proactor.o:(.rodata+0x0): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/POSIX_CB_Proactor.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/POSIX_Proactor.o:(.rodata+0x0): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/POSIX_Proactor.o:(.data.rel.ro+0xbc): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/Priority_Reactor.o:(.rodata+0x0): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/Priority_Reactor.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/Proactor.o:(.rodata+0x0): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/Proactor.o:(.data.rel.ro+0x9c): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/Reactor.o:(.rodata+0x10): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/Reactor.o:(.data.rel.ro+0xc): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/TP_Reactor.o:(.rodata+0x0): multiple definition of `typeinfo name for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.rodata+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/TP_Reactor.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >'; .shobj/Asynch_Pseudo_Task.o:(.data.rel.ro+0x0): first defined here
/usr/lib/gcc-cross/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/bin/ld: .shobj/FILE_Addr.o: in function `ACE_FILE_Addr::set(ACE_FILE_Addr const&)':
/home/host/cross/ACE_TAO/ACE/ace/OS_NS_stdlib.inl:285: Warnung:the use of `mktemp' is dangerous, better use `mkstemp' or `mkdtemp'
collect2: error: ld returned 1 exit status
make[1]: *** [/home/host/cross/ACE_TAO/ACE/include/makeinclude/rules.lib.GNU:245: libACE.so.7.0.2] Fehler 1
make[1]: Verzeichnis „/home/host/cross/ACE_TAO/ACE/ace“ wird verlassen
make: *** [GNUmakefile:262: ACE] Fehler 2
make: Verzeichnis „/home/host/cross/ACE_TAO/TAO“ wird verlassen
```

### Repeat by

I followed the Github workflow actions for linux ubuntu 18.04-g++8 and added the CROSS_COMPILE statement in the macros file. The command to build ACE_TAO returned with an error. The Linker is unable to create the shared objects.

### Sample fix/ workaround

Unknown

--- a/ACE/ace/Select_Reactor.h
+++ b/ACE/ace/Select_Reactor.h
@@ -22,7 +22,9 @@
 ACE_BEGIN_VERSIONED_NAMESPACE_DECL
 
 typedef ACE_Reactor_Token_T<ACE_SELECT_TOKEN> ACE_Select_Reactor_Token;
+#if defined ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION_EXPORT
 template class ACE_Export ACE_Select_Reactor_T<ACE_Select_Reactor_Token>;
+#endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION_EXPORT */
 
 typedef ACE_Select_Reactor_T<ACE_Select_Reactor_Token> ACE_Select_Reactor;
 

